src/runtime/objmodel.cpp

```diff
@@ -1705,8 +1711,37 @@ void setattrInternal(Box* obj, const std::string& attr, Box* val, SetattrRewrite
         // We don't need to to the invalidation stuff in this case.
         return;
     } else {
+        // Finally, check __setattr__
+        if (obj->cls->tp_setattr) {
+            rewrite_args = NULL;
+            REWRITE_ABORTED("");
+
+            int rtn = obj->cls->tp_setattr(obj, const_cast<char*>(attr.c_str()), val);
```

Comment:
Hmm I'd prefer to leave the const_cast explicit.  Taking a look at the standard library, any module that sets tp_setattr explicitly casts it to setattrfunc, so in theory we could try changing the setattrfunc typedef without the ifdef, but for a reason that defeats the type checking anyway.

I guess let's just keep this for now even though it's worrisome...